.github/actions/upload-nextjs-bundle-stats/action.yaml

```diff
@@ -0,0 +1,24 @@
+name: upload-nextjs-bundle-stats
+description: Upload NextJS bundle stats
+
+inputs:
+  path_to_nextjs_folder:
+    description: 'The path to reach the NextJS folder (e.g. ./apps/website)'
+    required: true
+  upload_path:
+    description: 'The path where the report should be uploaded'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Generate bundle report
+      working-directory: ${{ inputs.path_to_nextjs_folder }}
+      shell: bash
+      run: npx -p nextjs-bundle-analysis@0.5.0 report
+
+    - name: Upload bundle
+      uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+      with:
+        name: bundle
+        path: ${{ inputs.upload_path }}
```

Comment:
How long is the artifact kept?

Reply:
The retention period is 90 days (configured within repository's settings).

It is also possible to override that retention value with a custom configuration of the upload action (https://github.com/actions/upload-artifact#retention-period)